ux.symfony.com/src/Twig/Components/Animalz/FacetMenu.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Twig\Components\Animalz;
+
+use App\Enum\AnimalzType;
+use App\Service\AnimalzRepository;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveAction;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\ComponentToolsTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+
+#[AsLiveComponent(name: 'Animalz:FacetMenu', template: 'components/Animalz/FacetMenu.html.twig')]
+final class FacetMenu
+{
+    use ComponentToolsTrait;
+    use DefaultActionTrait;
+
+    private const array EMITTABLE_PROPS = ['name', 'type', 'maxLegs'];
+
+    #[LiveProp(writable: true, onUpdated: 'emitChange', url: true)]
+    public ?string $name = null;
+
+    #[LiveProp(writable: true, onUpdated: 'emitChange', url: true)]
+    public ?AnimalzType $type = null;
+
+    #[LiveProp(writable: true, onUpdated: 'emitChange', url: true)]
+    public ?int $maxLegs = null;
+
+    public function __construct(
+        private readonly AnimalzRepository $animalzRepository,
+    ) {
+    }
+
+    #[LiveAction]
+    public function reset(): void
+    {
+        $this->name = null;
+        $this->type = null;
+        $this->maxLegs = null;
+
+        $this->emitChange();
+    }
+
+    public function emitChange(): void
+    {
+        $this->nullTheProps();
+        $this->emit('facetSetted', $this->convertPropertiesToAssociativeArray());
```

Comment:
```suggestion
        $this->emit('Animalz.FacetMenu.facetSetted', $this->convertPropertiesToAssociativeArray());
```

perhaps be more precise to locate/listen particular event ?